src/test/java/com/booking/superride/controller/CustomerControllerTest.java

```diff
@@ -0,0 +1,28 @@
+package com.booking.superride.controller;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+
+import com.booking.superride.common.ContainersConfig;
+import com.booking.superride.domain.CustomerDetailsDTO;
+import io.restassured.http.ContentType;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
```

Comment:
you are loading container freshly for each controller test which is wrong approach, as all these are integration tests you should load spring container and even spin up db only once.